hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/container/states/ContainerEntry.java

```diff
@@ -42,14 +43,22 @@ public ContainerInfo getInfo() {
   }
 
   public Set<ContainerReplica> getReplicas() {
-    return new HashSet<>(replicas.values());
+    return Collections.unmodifiableSet(replicas.keySet());
```

Comment:
Since a new map is created for each change, we can create the unmodifiable wrapper on write instead.